src/main/java/org/taktik/icure/services/external/rest/v1/facade/DocumentFacade.java

```diff
@@ -158,7 +159,14 @@ public Response getAttachment(@PathParam("documentId") String documentId, @PathP
 			byte[] attachment = document.decryptAttachment(isBlank(enckeys) ? null : Arrays.asList(enckeys.split(",")));
 			if (attachment != null) {
 				byte[] finalAttachment = attachment;
-				response = ResponseUtils.ok((StreamingOutput) output -> {
+				UTI uti = UTI.get(document.getMainUti());
+
+				String mimeType = uti != null && uti.getMimeTypes().size()>0 ? uti.getMimeTypes().get(0) : "application/octet-stream";
+
+				response = Response.ok()
+						.header("Content-Type", mimeType)
```

Comment:
This will change the return type of the getAttachment function client-side, so special handling will be needed.